### ISSUE TITLE
TabularExplainer on SHAPIQ and KernelExplainer on SHAP extremely slow on TabPFN for small dataset (173 instances)

### ISSUE THREAD

**Issue by wasup07** [shap/shap#4194]
Hi @slundberg

I'm trying to use KernelExplainer (on SHAP) and TabularExplainer (on SHAPIQ) to explain TabPFN predictions with beeswarm plot, following the documentation examples. However, the computation is taking extremely long for a very small test set (173 instances) on my personal computer, and I had to interrupt the process.

Setup:

Model: TabPFN

Test set size: 173 instances

Explainer: TabularExplainer

Questions:

Is TabPFN inherently slow for SHAP explanations compared to XGBoost/LightGBM/Random Forest?

Are there any optimization parameters or alternative approaches for explaining TabPFN more efficiently?

What would be a reasonable expected runtime for this dataset size?

System specs: Intel(R) Core(TM) Ultra 9 185H (2.30 GHz) processor and 32 GB of RAM.

Thank you!

**Comment by CloseChoice** [comment]
Hey, can you pls point us to the dataset? Could it be that there are a lot of features in this dataset?

**Comment by wasup07** [comment]
> Hey, can you pls point us to the dataset? Could it be that there are a lot of features in this dataset?

Hi @CloseChoice 

Thank you for your reply. 

I have 13 features. As for the dataset, it is confidential as it is a company dataset and I cannot share it publicly.

**Comment by CloseChoice** [comment]
we  need a reproducible example for this, to figure out if this is something we can change. Furthermore improvements to the KernelExplainer are welcome but not on the roadmap of the maintainers.

Note: this is the issue tracker for SHAP not shapiq, which is a different repo with other maintainers. It is a fantastic package though.
